hack/generator/pkg/astbuilder/builder.go

```diff
@@ -195,3 +195,80 @@ func MakeMap(key ast.Expr, value ast.Expr) *ast.CallExpr {
 		},
 	}
 }
+
+// TypeAssert returns an assignment statement with a type assertion
+// 	<lhs>, ok := <rhs>.(<type>)
+func TypeAssert(lhs ast.Expr, rhs ast.Expr, typ ast.Expr) *ast.AssignStmt {
```

Comment:
I don't think that there are any places in these methods that actually use the parameters in a way that could trigger a null pointer panic.

For example, for this method, the parameters `lhs, rhs, typ` can all be nil and there is no panic. It may not generate valid code, but at least right now the builder isn't guaranteeing valid code anyway (you could have conflicting variable names, invalid syntax, etc).